File: SmartReceipts/PDF/RenderedPDFLabel.m
//
//  RenderedPDFLabel.m
//  SmartReceipts
//
//  Created by Jaanus Siim on 09/07/15.
//  Copyright (c) 2015 Will Baumann. All rights reserved.
//

#import "RenderedPDFLabel.h"

@implementation RenderedPDFLabel

- (void)drawLayer:(CALayer *)layer inContext:(CGContextRef)ctx {
    BOOL isPDF = !CGRectIsEmpty(UIGraphicsGetPDFContextBounds());
    if (!layer.shouldRasterize && isPDF) {
        [self drawRect:self.bounds]; // draw unrasterized
    } else {
        [super drawLayer:layer inContext:ctx];
    }
}

@end

File: SmartReceipts/Reports/PDF/RenderedPDFLabel.swift
//
//  RenderedPDFLabel.swift
//  SmartReceipts
//
//  Created by Bogdan Evsenev on 01/12/2017.
//  Copyright © 2017 Will Baumann. All rights reserved.
//

import UIKit

class RenderedPDFLabel: UILabel {
    
    override var text: String? {
        didSet {
            guard let new = text?.replacingOccurrences(of: NON_BREAKING_SPACE, with: " "), text != new else { return }
            text = new
        }
    }
    
    override func draw(_ layer: CALayer, in ctx: CGContext) {
        let isPDF = !UIGraphicsGetPDFContextBounds().isEmpty
        if !layer.shouldRasterize && isPDF {
            draw(bounds)
        } else {
            super.draw(layer, in: ctx)
        }
    }
}
